src/app/(app)/attendance-report/actions/attendance-report.ts

```diff
@@ -0,0 +1,88 @@
+"use server";
+
+import { AttendancePercentage, AttendanceProgress } from "@/constant/constant";
+import { getAttendance } from "../../attendance/actions/getAttendance";
+
+export const getAttendanceReport = async () => {
+  const attendanceData = await getAttendance();
+
+  if (!attendanceData) return [];
+
+  const getStatus = (percentage: number): AttendanceProgress => {
+    if (percentage > AttendancePercentage.Excellent) {
+      return AttendanceProgress.Excellent;
+    }
+    if (percentage > AttendancePercentage.VeryGood) {
+      return AttendanceProgress.VeryGood;
+    }
+    if (percentage > AttendancePercentage.Good) {
+      return AttendanceProgress.Good;
+    }
+    if (percentage > AttendancePercentage.Average) {
+      return AttendanceProgress.Average;
+    }
+    return AttendanceProgress.low;
+  };
+
+  // Get the first day of the month and today's date
+  const today = new Date();
+  const firstDayOfMonth = new Date(today.getFullYear(), today.getMonth(), 1);
+
+  const isWithinCurrentMonth = (dateString: string): boolean => {
+    const date = new Date(dateString);
+    return date >= firstDayOfMonth && date <= today;
+  };
+
+  const uniqueMembers = Array.from(
+    new Set(attendanceData.map((attendance) => attendance.memberId))
+  );
+
+  return uniqueMembers.map((memberId) => {
```

Comment:
where this will use i can not find any usage of this

Reply:
Fixed